### DISCUSSION TITLE
Parse output for object contains keys with undefined value. How to skip those keys ?

### DISCUSSION THREAD

**Discussion by xinha-sh** [open-circle/valibot#196] [Q&A]
```ts
import * as v from "valibot";

const baseSchema = v.object({
  x: v.optional(v.array(v.string())),
  y: v.optional(v.array(v.string())),
 z: v.optional(v.array(v.string())),
})

const schemaA = v.object({
  key1: v.optional(baseSchema), 
  key2: v.optional(baseSchema)
})

const input = {key1:  {x: ["hello", "world"]}}
const result = v.parse(schemaA, input)

// result.output
// {key1: {x:  ["hello", "world"]}, key2: undefined}
```

Why do I get ``key2: undefined`` in the output ? What's the best way to ignore them.

My use case is that upon successful parsing I want to loop over array values via ``Object.values(catalog).map(item => objectKeys(item)).flat()`` which breaks with error ``Cannot convert undefined or null to object``

This used to work as intended with zod schema. 
Unless my input is ``const input = {key1:  {x: ["hello", "world"]}, key2: undefined}`` the output shouldn't contain undefined.

What's valibot's way of stripping away undefined keys ?

**Comment by fabian-hiller** [comment]
Can you move this into a issue? Basically what you saying is, that you only want keys with `undefined` if they are provided.

**Reply by xinha-sh** [reply]
@fabian-hiller Yes. 

Sure I'll move this to issue.
